Lib/ensurepip/__init__.py

```diff
@@ -26,9 +27,17 @@ def _run_pip(args, additional_paths=None):
     if additional_paths is not None:
         sys.path = additional_paths + sys.path
 
-    # Install the bundled software
-    import pip._internal
-    return pip._internal.main(args)
+    # Invoke pip as if it's the main module, and catch the exit.
+    backup_argv = sys.argv[:]
+    sys.argv[1:] = args
+    try:
+        runpy.run_module("pip", run_name="__main__", alter_sys=True)
+    except SystemExit as e:
+        return e.code
+    finally:
+        sys.argv[:] = backup_argv
+
+    assert 0, "should never reach here"
```

Comment:
Why don't just write a return o raise an exception instead of the message "Should never reach here"?

Reply:
Well, they'd be functionally the same as far as I'm concerned. This is just a sanity-check, so I'm using an assert.

Anyway, if this is *the problem* with this PR, I'll be happy to change it.

Reply:
"raise SystemError" complaining that pip didn't exit as expected would be the way to go here.